README.md

```diff
@@ -73,23 +72,62 @@ ARGS:
     <paths>...    Sets the input paths to use
 ```
 
-Let's see a few examples, assuming the source directory is `~/Documents/mozilla-central` and the build directory is `~/Documents/mozilla-central/build`.
 
-### LCOV output
+## How to get grcov
+
+Grcov can be download grcov from [releases](https://github.com/mozilla/grcov/releases) or if you have rust installed,
+you can run `cargo install grcov`.
+
+## Example: How to generate .gcda files for from C/C++
+
+Pass `-ftest-coverage` and `-fprofile-arcs` options to `gcc` (see [gcc docs](https://gcc.gnu.org/onlinedocs/gcc/Gcov-Data-Files.html)), or pass the `--coverage` option to `clang`
+
+## Example: How to generate .gcda fiels for a Rust project
+
+1. Ensure that the following environment variables are set up:
 
 ```sh
-grcov ~/Documents/mozilla-central/build -t lcov > lcov.info
+export CARGO_INCREMENTAL=0
+export RUSTFLAGS="-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zno-landing-pads"
 ```
+These will ensure that things like dead code elimination do not skew the coverage.
+
+2. Build your code:
+
+`cargo build`
+
+If you look in `target/debug/deps` dir you will see `.gcno` files have appeared. These are the locations that could be covered.
+
+3. Run your tests:
+
+`cargo test`
+
+In the `target/debug/deps/` dir you will now also see `.gcda` files. These contain the hit counts on which of those locations have been reached. Both sets of files are used as inputs to `grcov`. 
 
-As the LCOV output is compatible with `lcov`, `genhtml` can be used to generate a HTML summary of the code coverage:
+## Genrate a coverage report from .gcda files
+
+Generate a html coverage report like this:
 ```sh
-genhtml -o report/ --show-details --highlight --ignore-errors source --legend lcov.info
+grcov ./target/debug/ -s . -t html --llvm --branch --ignore-not-existing -o ./target/debug/coverage/
+```
+
+You can see the report in `target/debug/coverage/index.html`.
+
+(or alterntatively with `-t lcov` grcov will output a lcov compatible coverage report that you could then feed into lcov's `genhtml` command).
+
+### lcov's genhtml
+
+By passing `-t lcov` you could generate an lcov.info file and pass it to genhtml:
+```sh
+genhtml -o ./target/debug/coverage/ --show-details --highlight --ignore-errors source --legend ./target/debug/lcov.info
 ```
 
 ### Coveralls/Codecov output
 
+Coverage can also be sent directly from grcov to coveralls:
```

Comment:
It is not really sent, just generated in a format compatible with those services